pkg/connection/credentials.go

```diff
@@ -0,0 +1,41 @@
+package connection
+
+// Credentials interface is to be implemented by any struct that knows how to
+// save/restore credentials for a given system.
+type Credentials interface {
```

Comment:
I am not entirely sure about this because this interface shouldn't be the source of truth on the expiration of such credentials. Rather, it's SCC who knows if a given credential is valid or not. Hence, a system that stores credentials in whatever way, assumes that the credentials it has are correct (i.e. not expired), until a request fails or the user manually removes said credentials.

Checking whether credentials have expired should be delivered through connections, and reacting to bad authentication. In the case of SUSEConnect, for example, an authentication error (by using expired credentials) means that it needs to tell the user to register again.

Reply:
Got it.